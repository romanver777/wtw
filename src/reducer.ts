import movies from "./mocks/films.json";
import { FilmsType } from "./types/types";
import { GENRES } from "./helpers/const";

interface InitStateType {
  films: FilmsType;
  genre: string;
}

interface ActionType {
  SET_GENRE: string;
  GET_FILMS_BY_GENRE: string;
}

const InitialState: InitStateType = {
  films: [...movies],
  genre: GENRES[0],
};

const ActionType = {
  SET_GENRE: `SET_GENRE`,
  GET_FILMS_BY_GENRE: `GET_FILMS_BY_GENRE`,
};

const ActionCreator = {
  setGenre: (genre: string) => ({
    type: ActionType.SET_GENRE,
    payload: genre,
  }),
  getFIlmsByGenre: (genre, films) => ({
    type: ActionCreator.GET_FILMS_BY_GENRE,
    payload: [],
  }),
};

const reducer = (state = InitialState, action): InitStateType => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload,
      };
    default:
      return state;
  }
};

const Operation = {};

export { reducer, ActionCreator, Operation };
