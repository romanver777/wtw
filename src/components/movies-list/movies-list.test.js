/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { reducer } from "../../reducer";
import { Router } from "react-router";
import history from "../../history";

import MoviesList from "./movies-list";
import films from "../../mocks/films.json";

describe("MoviesList component", () => {
  const initState = {
    films,
    awaitFilm: null,
    currentFilm: null,
    hoveredFilm: null,
    activeGenre: "",
    genresList: null,
    tk: null,
    reviews: null,
    staff: null,
    hoveredVideoData: null,
  };
  const store = createStore(reducer, initState);

  describe("no video", () => {
    const props = {
      films,
      handleClick: () => {},
      isShowMore: false,
    };
    const Trenderer = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <MoviesList {...props} />
        </Router>
      </Provider>
    );

    it("renders correctly", () => {
      expect(Trenderer.toJSON()).toMatchSnapshot();
    });
    it("don`t render ShowMore button", () => {
      const className = "catalog__more";
      expect(Trenderer.root.findAllByProps({ className })).toEqual([]);
    });
  });

  describe("with video", () => {
    const props = {
      films,
      handleClick: () => {},
      isShowMore: true,
    };
    const Trenderer = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <MoviesList {...props} />
        </Router>
      </Provider>
    );

    it("renders correctly", () => {
      expect(Trenderer.toJSON()).toMatchSnapshot();
    });
    it("renders ShowMore button", () => {
      const className = "catalog__more";
      expect(Trenderer.root.findAllByProps({ className })).toHaveLength(1);
    });
  });
});
