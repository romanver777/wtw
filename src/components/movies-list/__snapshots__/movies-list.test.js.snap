// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component no video renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Соединённые Штаты против Билли Холидей"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1302121"
        onClick={[Function]}
      >
        Соединённые Штаты против Билли Холидей
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Неуязвимый (сериал)"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1171895.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1171895"
        onClick={[Function]}
      >
        Неуязвимый (сериал)
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Невероятные (сериал)"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1245486"
        onClick={[Function]}
      >
        Невероятные (сериал)
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Райя и последний дракон"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1015471.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1015471"
        onClick={[Function]}
      >
        Райя и последний дракон
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Бесстыжие (сериал)"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/571335.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/571335"
        onClick={[Function]}
      >
        Бесстыжие (сериал)
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Happy End (сериал)"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1328036.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1328036"
        onClick={[Function]}
      >
        Happy End (сериал)
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Конёк-Горбунок"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1045583.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1045583"
        onClick={[Function]}
      >
        Конёк-Горбунок
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Родные"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1274271.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1274271"
        onClick={[Function]}
      >
        Родные
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Фарго (сериал)"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/767379.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/767379"
        onClick={[Function]}
      >
        Фарго (сериал)
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Зависнуть в Палм-Спрингс"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1245524.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1245524"
        onClick={[Function]}
      >
        Зависнуть в Палм-Спрингс
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Абсолютное доказательство"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/4430018.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/4430018"
        onClick={[Function]}
      >
        Абсолютное доказательство
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Чикатило (сериал)"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1437860.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1437860"
        onClick={[Function]}
      >
        Чикатило (сериал)
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Майор Гром"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1005008.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1005008"
        onClick={[Function]}
      >
        Майор Гром
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="В её глазах (мини-сериал)"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1291310.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1291310"
        onClick={[Function]}
      >
        В её глазах (мини-сериал)
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Охотник на монстров"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1038789.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1038789"
        onClick={[Function]}
      >
        Охотник на монстров
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Аферистка"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1248786.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1248786"
        onClick={[Function]}
      >
        Аферистка
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Грань времени"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1193013.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1193013"
        onClick={[Function]}
      >
        Грань времени
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Охотник за разумом (сериал)"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/958500.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/958500"
        onClick={[Function]}
      >
        Охотник за разумом (сериал)
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Звездный рубеж"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1347194.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1347194"
        onClick={[Function]}
      >
        Звездный рубеж
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Ассасин: Битва миров"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/1236082.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#/film/1236082"
        onClick={[Function]}
      >
        Ассасин: Битва миров
      </a>
    </h3>
  </article>
</div>
`;

exports[`MoviesList component with video renders correctly 1`] = `
Array [
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Соединённые Штаты против Билли Холидей"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1302121"
          onClick={[Function]}
        >
          Соединённые Штаты против Билли Холидей
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Неуязвимый (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1171895.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1171895"
          onClick={[Function]}
        >
          Неуязвимый (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Невероятные (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1245486"
          onClick={[Function]}
        >
          Невероятные (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Райя и последний дракон"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1015471.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1015471"
          onClick={[Function]}
        >
          Райя и последний дракон
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Бесстыжие (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/571335.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/571335"
          onClick={[Function]}
        >
          Бесстыжие (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Happy End (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1328036.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1328036"
          onClick={[Function]}
        >
          Happy End (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Конёк-Горбунок"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1045583.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1045583"
          onClick={[Function]}
        >
          Конёк-Горбунок
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Родные"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1274271.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1274271"
          onClick={[Function]}
        >
          Родные
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Фарго (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/767379.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/767379"
          onClick={[Function]}
        >
          Фарго (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Зависнуть в Палм-Спрингс"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1245524.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1245524"
          onClick={[Function]}
        >
          Зависнуть в Палм-Спрингс
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Абсолютное доказательство"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/4430018.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/4430018"
          onClick={[Function]}
        >
          Абсолютное доказательство
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Чикатило (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1437860.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1437860"
          onClick={[Function]}
        >
          Чикатило (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Майор Гром"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1005008.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1005008"
          onClick={[Function]}
        >
          Майор Гром
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="В её глазах (мини-сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1291310.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1291310"
          onClick={[Function]}
        >
          В её глазах (мини-сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Охотник на монстров"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1038789.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1038789"
          onClick={[Function]}
        >
          Охотник на монстров
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Аферистка"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1248786.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1248786"
          onClick={[Function]}
        >
          Аферистка
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Грань времени"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1193013.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1193013"
          onClick={[Function]}
        >
          Грань времени
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Охотник за разумом (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/958500.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/958500"
          onClick={[Function]}
        >
          Охотник за разумом (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Звездный рубеж"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1347194.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1347194"
          onClick={[Function]}
        >
          Звездный рубеж
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Ассасин: Битва миров"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1236082.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1236082"
          onClick={[Function]}
        >
          Ассасин: Битва миров
        </a>
      </h3>
    </article>
  </div>,
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Больше
    </button>
  </div>,
]
`;
